diff --git a/tools_webrtc/libs/generate_licenses.py b/tools_webrtc/libs/generate_licenses.py
index d945b43cca..f09ed3702b 100755
--- a/tools_webrtc/libs/generate_licenses.py
+++ b/tools_webrtc/libs/generate_licenses.py
@@ -86,6 +86,11 @@ LIB_TO_LICENSES_DICT = {
     # Compile time dependencies, no license needed:
     'ow2_asm': [],
     'jdk': [],
+
+    'catapult': [],
+    'google_benchmark': [],
+    'googletest': [],
+    'vinn': [],
 }
 
 # Third_party library _regex_ to licences mapping. Keys are regular expression
 